apps/grc-platform/backend/internal/middleware/guard_test.go

```
@@ -0,0 +1,322 @@
+		IdPs:                  []config.IdPConfig{idpCfg(guardIssuer, "api")},
+		TestKeyFuncs:          map[string]jwt.Keyfunc{guardIssuer: testKeyFunc},
+		Router:                mux,
+		InternalEmailDomains:  domains,
+	}
+}
+
+// serveGuarded runs one request through Auth and returns the status code.
+// Auth wraps cfg.Router itself, matching production, so an allowed request is
+// answered by the real mux — including its own 404/405.
+func serveGuarded(cfg middleware.Config, method, path, token string) int {
+	var next http.Handler = okHandler()
```

golangci-lint (2.12.2)

[error] ST1023: should omit type http.Handler from declaration; it will be inferred from the right-hand side

(staticcheck)